main.py

```diff
@@ -53,13 +53,17 @@ def analyze_messages(inbox_path, number_conversations):
     )
     plt.yticks(size=8)
     plt.xticks(size=8, rotation=25, ha="right")
-    plt.bar(
+    bars = plt.bar(
         nlargest_messages[0],
         nlargest_messages[1],
         width=0.7,
         color=(0.1, 0.1, 0.1, 0.1),
         edgecolor="blue",
     )
+    for bar in bars:
+        yval = bar.get_height()
+        plt.text(bar.get_x() + bar.get_width()/2, yval * 1.015, yval, ha='center', size=8)
```

Comment:
I think it makes more sense to have a constant y-offset for the height of the label. Something like `yval + 10` instead of `yval * 1.015`.

Reply:
Would it be possible to get the label to format with thousands separators?

Reply:
I tried to use a constant y-offset but it wasn't working regardless if I used yval + 1, yval + 10, or yval + 100.

Reply:
What do you by it wasn't working? What are the units of the height argument? I thought it would be pixels, but thinking about it some more it's probably the actual y-value. So, for example, you're looking at `yval = 100,000` so adding 100 isn't going to do much. I would probably add 2(ish)% of the max y-value.

Reply:
I tried to use a constant y-offset but it wasn't working regardless if I used ```yval + 1```, ```yval + 10```, or ```yval + 100```.

Reply:
I think something like this should work and would allow for a constant offset for all labels.

```python
    y_offset = max(nlargest_messages[1]) * 0.015
    for bar in bars:
        yval = bar.get_height()
        plt.text(
            bar.get_x() + bar.get_width() / 2,
            yval + y_offset,
            yval,
            ha="center",
            size=8,
        )
```

Reply:
And changing the third argument from `yval` to `f"{yval:,}"` will add thousands separators.

Reply:
> I think something like this should work and would allow for a constant offset for all labels.
> 
> ```python
>     y_offset = max(nlargest_messages[1]) * 0.015
>     for bar in bars:
>         yval = bar.get_height()
>         plt.text(
>             bar.get_x() + bar.get_width() / 2,
>             yval + y_offset,
>             yval,
>             ha="center",
>             size=8,
>         )
> ```

Oh, I understand what you are saying. Good solution.